File: src/main/java/com/jacuum/algo/RobotAlgorithm.java
package com.jacuum.algo;

import org.springframework.core.annotation.AliasFor;
import org.springframework.stereotype.Component;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Marks a {@link RobotAlgo} implementation as auto-discoverable.
 * The {@code value} is the display name shown in the UI.
 * If omitted, the simple class name is used.
 */
@Component
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface RobotAlgorithm {
    @AliasFor(annotation = Component.class)
    String value() default "";
}

File: src/main/kotlin/com/jacuum/algo/RobotAlgorithm.kt
package com.jacuum.algo

import org.springframework.core.annotation.AliasFor
import org.springframework.stereotype.Component

@Component
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class RobotAlgorithm(
    @get:AliasFor(annotation = Component::class)
    val value: String = ""
)
